velox/dwio/common/tests/utils/BatchMaker.cpp

```diff
@@ -184,11 +184,19 @@ VectorPtr BatchMaker::createVector<TypeKind::HUGEINT>(
     MemoryPool& pool,
     std::mt19937& gen,
     std::function<bool(vector_size_t /*index*/)> isNullAt) {
+  int bitsToMove = 0;
+  // Generate proper bits of random value for LongDecimalType tests.
+  if (type->isLongDecimal()) {
+    auto [precision, scale] = getDecimalPrecisionScale(*type);
+    int bitsNeeded = ceil(log2(std::pow(10, precision))) + 1;
```

Comment:
why + 1? 
also, strictly speaking, the maximum #digits `precision` can express is std::pow(10, precision) - 1.
SHould it be `ceil(log2(std::pow(10.0, precision) - 1));`?

Reply:
This is just a rough calculation. I don't know if there is a better way to do this calculation. But in the case it is DECIMAL(38), we want it to create data with 128 digits, right? In that case, `ceil(log2(std::pow(10, precision))) + 1;` is equal to 128.

Reply:
The 1 extra digit (the most significant digit (MSD)) I think is for the +/- sign.